packages/core/src/amazonq/commons/controllers/contentController.ts

```diff
@@ -30,4 +31,24 @@ export class EditorContentController {
                 )
         }
     }
+
+    public async acceptDiff(message: any) {
+        const { filePath } = message?.context?.activeFileContext || {}
+        if (filePath && message?.code?.trim().length > 0) {
+            const right = Uri.file(filePath) // Existing File to Write
+            const fileName = path.basename(right.path)
+
+            const diffScheme = 'amazon-q-accept-diff'
+            const left = Uri.parse(`${diffScheme}:${fileName}-` + randomUUID())
```

Comment:
`getNonexistentFilename` may be useful https://github.com/aws/aws-toolkit-vscode/blob/1c1b0bfa7600d78cb73debcc82da1f9700047215/packages/core/src/shared/filesystemUtilities.ts#L189